jinahub/indexers/dbms/LMDBDBMSIndexer/__init__.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Dict
+
+from jina import Executor, Document, DocumentArray, requests
+
+import lmdb
+
+
+class LMDBDBMSIndexer(Executor):
+    """An lmbd-based DBMS Indexer for Jina
+
+    For more information on lmdb check their documentation: https://lmdb.readthedocs.io/en/release/
+    """
+
+    def __init__(
+        self,
+        map_size: int = 10485760,
+        default_traversal_path: str = 'r',
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.default_traversal_path = default_traversal_path
+        self.file = os.path.join(self.workspace, 'db.lmdb')
+        if not os.path.exists(self.workspace):
+            os.makedirs(self.workspace)
+        # db for internal usage
+        self.db = 'db'
+        # see https://lmdb.readthedocs.io/en/release/#environment-class for usage
+        self.env = lmdb.Environment(
+            self.file,
+            map_size=map_size,
+            subdir=False,
+            readonly=False,
+            metasync=True,
+            sync=True,
+            map_async=False,
+            mode=493,
+            create=True,
+            readahead=True,
+            writemap=False,
+            meminit=True,
+            max_readers=126,
+            max_dbs=0,  # means only one db
+            max_spare_txns=1,
+            lock=True,
+        )
+
+    @requests(on='/index')
+    def index(self, docs: DocumentArray, parameters: Dict = None, **kwargs):
```

Comment:
I think by definition `parameters` is not Optional, but however, if we want to make it so, at least let's have proper annotation

Reply:
We can't force a user to pass parameters, I think. Handling it with allowing it to be optional allows us to catch it anyway.